src/main/java/hudson/plugins/openid/OpenIdExtension.java

```diff
@@ -107,11 +107,16 @@ public void extendFetch(FetchRequest request) throws MessageException{}
      *
      * @param authRequest the authentication request.
      * @throws MessageException if there is a message error extending the request
+     * @throws IllegalStateException if there is no Jenkins instance
      */
-    public static void extendRequest(AuthRequest authRequest) throws MessageException {
+    public static void extendRequest(AuthRequest authRequest) throws MessageException, IllegalStateException {
         FetchRequest request = FetchRequest.createFetchRequest();
+        Jenkins jenkins = Jenkins.getInstance();
```

Comment:
This code is duplicated multiple times. Better to use a Restricted utility method